validator/src/validator.rs

```diff
@@ -630,25 +645,31 @@ where
         self.consensus.is_ready_for_validation()
     }
 
-    fn get_staking_state(&self, blockchain: &Blockchain) -> ValidatorStakingState {
+    fn get_validator(&self, blockchain: &Blockchain) -> Result<Option<ValidatorAccount>, ()> {
```

Comment:
Nope. Unknown is when we fail to read the staking contract. When it's none, it returns Inactive.

Reply:
Now if `get_validator` returns `Err(())`, it means that the `staking_contract` can't be found. That feels weird. You can only know this by reading the function. Result should then return a proper Error here instead of `()`.

Looking at it again you can even get rid of `ValidatorStakingState::NoStake` as it's a dead variant. Then it can be `get_validator -> Option<ValidatorAccount>`